src/shogun/optimization/L1Penalty.h

```diff
@@ -60,15 +60,36 @@ class L1Penalty: public SparsePenalty
 	 */
 	virtual float64_t get_penalty(float64_t variable) {return CMath::abs(variable);}
 
+
+
+	/** Return the gradient of the penalty wrt a target variable
+	 *
+	 * Note that for L1 penalty we do not compute the gradient/sub-gradient in our implementation.
+	 * Instead, we do a proximal projection.
+	 *
+	 * @param variable value of a target variable
+	 * @param gradient_of_variable unregularized/unpenalized gradient of the variable
+	 * @return the gradient of the penalty wrt the variable
+	 */
+
 	virtual float64_t get_penalty_gradient(float64_t variable,
 		float64_t gradient_of_variable) {return 0.0;}
 
+	/** Set the rounding eplison
+	 *
+	 * @param eplison rounding eplison
```

Comment:
thanks!
corrected the typo at 
https://github.com/shogun-toolbox/shogun/pull/2956